Read pipe cleanup refactor

Aryex opened the pull request:
### Summary

* Extracted read pipe cleanup logic into a separate class. It is intended to be re-used by JSON reader. 
* Added new supporting interfaces. 
* Simplified existing partition interfaces:
  * Previously, ParquetFileRange and VerticaDistributedFileSystem define fields which are not needed to be optional. They are refactored to normal fields. Related files are updated.

### Related Issue

#416 

### Additional Reviewers
@alexey-temnikov 
@alexr-bq 
@jeremyp-bq 
@jonathanl-bq

jonathanl-bq reviewed (commented):
I don't have time to do a really thorough review, but I think we should avoid exposing index math in our traits. It feels like a lower level implementation detail more than something we should expose in a trait, but I don't know if there's a better justification for this current design.

Aryex commented:
> I don't have time to do a really thorough review, but I think we should avoid exposing index math in our traits. It feels like a lower level implementation detail more than something we should expose in a trait, but I don't know if there's a better justification for this current design.

Updated. Also changed the names of the traits to Cleanup and PartitionCleanup

jeremyp-bq commented on connector/src/main/scala/com/vertica/spark/datasource/partitions/PartitionCleanup.scala:
```diff
@@ -0,0 +1,22 @@
+import org.apache.spark.sql.connector.read.InputPartition
+
+trait PartitionCleanup extends InputPartition {
+  def getCleanUps: Seq[Cleanup]
```
Minor nit: getCleanUps when we use Cleanup (up not capitalized) everywhere else.

Aryex replied:
Updated

Aryex commented:
Updated trait name to better reflect the implementor capabilities
Cleanup -> PortionId: Implementors can identify themselves amongst other portions from the same file.
PartitionCleanup -> Cleanup: Implementors contains cleanup information.
